lib/msf/base/sessions/mainframe_shell.rb

```diff
@@ -73,37 +72,36 @@ def shell_read(length=-1, timeout=1)
   #
   def shell_write(buf)
     #mfimpl
-    return if not buf
+    return unless buf
 
     begin
       framework.events.on_session_command(self, buf.strip)
       rstream.write(Rex::Text.to_ibm1047(buf))
     rescue ::Rex::SocketError, ::EOFError, ::IOError, ::Errno::EPIPE => e
-      #print_error("Socket error: #{e.class}: #{e}")
       shell_close
       raise e
     end
   end
 
-  #
   def execute_file(full_path, args)
     #mfimpl
     raise NotImplementedError
   end
 
-  #
   def process_autoruns(datastore)
```

Comment:
@wvu-r7 
Oustanding! Everything works great except for the following:

For some reasonwith the `raise NotImplementedError` in `process_autoruns` the session hangs during initial creation.  If you CTRL+C it then `sessions -i <session_num> ` it works just fine from then on.

Since I haven't fully regression tested process_autoruns and really it's up to the payload if it wants autoruns as an option, I'd propose one of two things for this code in the mainframe_shell class.

1) something like this (which works great) in place of the `def process_autoruns(datastore)` method in `mainframe_shell`
```
# need to do more testing on this before we either use the
# default in command_shell or write a new one.  For now we
# just make it unavailble
undef_method  :process_autoruns
```
This works because then it does not respond to the following code in `msf/core/handler.rb`
```
    # Process the auto-run scripts for this session
    if session.respond_to?('process_autoruns')
      session.process_autoruns(datastore)
    end 
```
2) remove the `raise NotImplementedError` line and just leave the method empty, or return nil

I think the top solution is cleaner.

The top-level process_autoruns in command_shell.rb works for the few test cases I've thrown it, but given the if..then with "force_encoding" etc and the care I have to have around characterset translations, I'd prefer to leave it undefined or nil for now.  

Thoughts?